app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+    
+  end
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+    user = User.find_or_create_from_omniauth(auth_hash)
+    if user
+      session[:user_id] = user.id
+      redirect_to root_path
+    else
+      redirect_to root_path
+    end
```

Comment:
To DRY your code better, you could use "redirect_to root_path" only once! and be a better-lazy-programmer :) 
 and replace lines (9-14) with only two lines of code:

      session[:user_id] = user.id if user 
      redirect_to root_path